src/main/java/appeng/me/cache/CraftingGridCache.java

```diff
@@ -426,6 +426,7 @@ public ImmutableCollection<ICraftingPatternDetails> getCraftingFor(
                 ImmutableList.Builder<ICraftingPatternDetails> allPatterns = ImmutableList.builder();
                 for (IAEItemStack alternative : substitutions) {
                     allPatterns.addAll(this.craftableItems.get(alternative));
+                    break;
```

Comment:
why break here?

Reply:
Yeah this would break substitution crafting by only checking one substitution

Reply:
This is not resolved yet

Reply:
Agreed. The break does not make sense.

Reply:
After discussion with author in DM, he mentions that the current crafting simulator assumes getCraftingFor() will return patterns that return one same kind of item. returning patterns for different item will lead to bug in simulator later on. This break is a bandaid fix for this. We might clear this up later, but for now this break is necessary.

Reply:
This sort of thing needs to be put in a comment